### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by hoangtrvu**
:v:

/cc @zendesk/vegemite

### Description
This PR is mainly for improving our current tests. Other than that, it also remove some unused variables.

### Tasks
- [x] Remove unused variables
- [x] Add additional test cases
- [x] Make tests self-contained and independent from fixtures
- [x] Fix tests that always pass
- [x] Fix tests that might pass/fail depending on their execution order

### Risks
* [LOW] Only remove unused code and update test files

### CODE REVIEW

File: src/steps/migrate_translations.ts [hunk 1]
```diff
@@ -3,5 +3,10 @@ export default async options => {
   const dest = options.get("dest");
   const editor = options.get("editor");
   // Copy all translation files across
-  editor.copy(`${src}/translations/*`, `${dest}/src/translations/`);
+  try {
```
Comment (svizzari) [inline src/steps/migrate_translations.ts]: I wonder whether it's worth extracting a spec helper for these copy operations, something like `tryCopy` that will wrap the `try... catch`?  It only appears 3 times, so not a huge deal, but could be slightly DRY'er

File: src/test/steps/migrate_manifest.test.ts [hunk 2]
```diff
@@ -2,65 +2,68 @@ import { expect } from "chai";
...
+    mkdir("-p", src);
+    mkdir("-p", dest);
     editor = fsEditor.create(memFs.create());
     options = Map({ src, dest, editor });
+    editor.writeJSON(srcManifestPath, {});
+  });
+
+  afterEach(() => {
+    rm("-rf", src);
+    rm("-rf", dest);
```
Comment (svizzari) [inline src/test/steps/migrate_manifest.test.ts]: #nit Should actually be able to consolidate this with the line above, i.e. https://github.com/shelljs/shelljs#rmoptions-file_array

File: src/test/steps/migrate_translations.test.ts [hunk 3]
```diff
@@ -1,23 +1,37 @@
...
+    mkdir("-p", src);
+    mkdir("-p", dest);
     editor = fsEditor.create(memFs.create());
     options = Map({ src, dest, editor });
   });
 
-  it("should copy translations to the destination", async () => {
+  afterEach(() => {
+    rm("-rf", src);
+    rm("-rf", dest);
```
Comment (svizzari) [inline src/test/steps/migrate_translations.test.ts]: See my previous comment re: consolidating to one line

File: src/steps/migrate_translations.ts (hunk 1 above)
Comment (hoangtrvu) [reply src/steps/migrate_translations.ts]: agree. Its usage will likely to increase in the future so I'll extract it to `utils.ts`.

**Comment by hoangtrvu** [comment]
PR updated based on @svizzari's comments

**Comment by svizzari** [comment]
> PR updated based on @svizzari's comments

Great, thanks @hoangtrvu.  My 👍 stands

**Pull request merged**